components/tidb_query_vec_executors/src/slow_hash_aggr_executor.rs

```diff
@@ -288,8 +285,14 @@ impl<Src: BatchExecutor> AggregationExecutorImpl<Src> for SlowHashAggregationImp
                         )?;
                         self.group_key_offsets.push(self.group_key_buffer.len());
                     }
-                    // we have checked that group by cannot be a scalar
-                    _ => unreachable!(),
+                    RpnStackNode::Scalar { value, field_type } => {
+                        value.as_scalar_value_ref().encode_sort_key(
+                            field_type,
+                            context,
+                            &mut self.group_key_buffer,
+                        )?;
```

Comment:
optional optimization: we can encode once and copy multiple times, instead of encode multiple times.